### PR TITLE
Spark 3.5: Remove obsolete conf parsing logic

### PR DESCRIPTION
**Pull request by aokolnychyi**
This PR removes some conf parsing logic that predates the split of Spark 2 and Spark 3.

### CODE REVIEW

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java [hunk 1]
```diff
@@ -70,6 +70,14 @@ public DurationConfParser durationConf() {
     return new DurationConfParser();
   }
 
+  private static CaseInsensitiveStringMap asCaseInsensitiveStringMap(Map<String, String> map) {
```
Comment (aokolnychyi) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java:73]: Prior to split of Spark 2 and Spark 3, we had to use `Map<String, String>` as Spark 3 used one class for options and Spark 2 used another one. It is always `CaseInsensitiveStringMap` in Spark 3 but I am checking here to be safe.

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java [hunk 2]
```diff
@@ -220,14 +228,10 @@ public ThisT tableProperty(String name) {
     }
 
     protected T parse(Function<String, T> conversion, T defaultValue) {
-      if (!optionNames.isEmpty()) {
```
Comment (aokolnychyi) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java:223]: The if here is redundant. We initially had `!= null` check before we added a list of options that is never null.

File: spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java [hunk 3]
```diff
@@ -220,14 +228,10 @@ public ThisT tableProperty(String name) {
...
-        for (String optionName : optionNames) {
-          // use lower case comparison as DataSourceOptions.asMap() in Spark 2 returns a lower case
-          // map
-          String optionValue = options.get(optionName.toLowerCase(Locale.ROOT));
-          if (optionValue != null) {
-            return conversion.apply(optionValue);
-          }
+      for (String optionName : optionNames) {
+        String optionValue = options.get(optionName);
```
Comment (aokolnychyi) [inline spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java:232]: Removed the explicit conversion to lower case and obsolete comment.

**Review by nastra — APPROVED** [review]

**Comment by aokolnychyi** [comment]
Thanks, @nastra!

**Pull request merged**